### PR TITLE
SER-2797 controller to get IP of the wiki

### PR DESCRIPTION
**Pull request by ghost**
h4.LINKS
* https://wikia-inc.atlassian.net/browse/SER-2797

create controller in MW to get IP of the wiki

### CODE REVIEW

File: extensions/wikia/WikiFactory/tests/WikiFounderIpControllerTest.php [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+		$dbw->insert( "city_list", [
+				"city_id" => self::WIKI_ID,
+				"city_url" => "http://test.wikia.com",
+				"city_sitename" => "wikdia",
+				"city_vertical" => 5,
+				"city_cluster" => "c1",
+				"city_dbname" => "unittests",
+				"city_founding_ip_bin" => inet_pton( "149.6.28.163" ),
+			], __METHOD__ );
```
Comment (macbre) [inline extensions/wikia/WikiFactory/tests/WikiFounderIpControllerTest.php]: Please move this row to a YAML fixture, e.g. - https://github.com/Wikia/app/blob/2a4a827f0ace3c942493e0db42d669656961aef7/extensions/wikia/WikiFactory/tests/fixtures/api_variable.yaml#L2

File: extensions/wikia/WikiFactory/api/WikiFounderIpController.php [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+	 * @throws WikiaHttpException
+	 */
+	public function assertCanAccessController() {
+		if ( !$this->getContext()->getRequest()->isWikiaInternalRequest() ) {
+			throw new ForbiddenException( 'Access to this controller is restricted' );
+		}
+		if ( $this->getContext()->getRequest()->wasPosted() ) {
+			throw new MethodNotAllowedException( 'Only GET allowed' );
+		}
+	}
```
Comment (macbre) [inline extensions/wikia/WikiFactory/api/WikiFounderIpController.php:37]: You can use `allowsExternalRequests` method instead - https://github.com/Wikia/app/blob/3a425cfe7d721bd0684f73b37988e4136c062c2c/extensions/wikia/Privacy/RemoveUserDataController.php#L12-L14

File: extensions/wikia/WikiFactory/tests/WikiFounderIpControllerTest.php [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+		$this->wikiFounderIpController->init();
+	}
+
+	public function testGetWikiIp() {
+		$fauxRequest = new FauxRequest( [ 'wikiId' => static::WIKI_ID ] );
+		$fauxRequest->setHeader( \Wikia\Tracer\WikiaTracer::INTERNAL_REQUEST_HEADER_NAME, 1 );
+		$this->requestContext->setRequest( $fauxRequest );
+
+		$wikiId = $fauxRequest->getVal( 'wikiId' );
+		self::assertThat( $wikiId, self::equalTo( static::WIKI_ID ) );
```
Comment (macbre) [inline extensions/wikia/WikiFactory/tests/WikiFounderIpControllerTest.php]: Why not `$this->assertEquals` here ?

File: extensions/wikia/WikiFactory/api/WikiFounderIpController.php [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+	/**
+	 * Make sure to only allow GET methods.
+	 * @throws WikiaHttpException
+	 */
+	public function assertCanAccessController() {
+		if ( $this->getContext()->getRequest()->wasPosted() ) {
+			throw new MethodNotAllowedException( 'Only GET allowed' );
+		}
```
Comment (macbre) [inline extensions/wikia/WikiFactory/api/WikiFounderIpController.php]: Do we really need to check the HTTP method used?

Comment (ghost) [reply extensions/wikia/WikiFactory/api/WikiFounderIpController.php]: You are right, when I don't check it and try POST the resource it is smart enough to return 400 automatically.

**Review by macbre — APPROVED** [review]

**Pull request merged**